### ISSUE TITLE
Request for Instance Normalization

### ISSUE THREAD

**Issue by crcrpar** [chainer/chainer#5229]
Instance Normalization is similar to BatchNormalization though it has been used in Style Transfer, Image Translation, switchable normalization, and IBN-Net.

**Comment by crcrpar** [comment]
I'm considering accessing this and have one question and one draft of the impl.

1. How to calculate/manage running mean and variance?
Usually, even in inference/testing mode, Instance Normalization normalizes inputs using their statistics as in the training. 
However, considering consistency with Batch Normalization, I think it better to manage running statistics even if they are not correctly calculated.
And if managing running statistics, how to handle arbitrary batch size?

**Comment by MannyKayy** [comment]
Hi, doesn't the [axis](https://github.com/chainer/chainer/pull/4266) option in the BatchNormalization link implicitly allow for Instance Normalization? Given a 4 dimensional input x, with  (B, C, H, W) = x.shape, if you set, axis to (3,4) it should be equivalent to instance normalization .

**Comment by crcrpar** [comment]
Can we set the shape of gamma, beta, running_mean, and running_var to (C,) by giving axis (2, 3) to the latest `L.BatchNormalization`?

**Comment by crcrpar** [comment]
Quick look at Batch Normalization Function and I noticed that we cannot enojoy cundnn Batch Normalization when len(gamma) > 1.
Am I correct?

**Comment by crcrpar** [comment]
Since this is a bit old and I changed the design, close this and submit a new PR.

**Comment by crcrpar** [comment]
bump

**Comment by hvy** [comment]
Reopening as PR is still ongoing.
